drivers/include/mtd.h

```diff
@@ -96,6 +110,13 @@ typedef struct {
     uint32_t sector_count;     /**< Number of sector in the MTD */
     uint32_t pages_per_sector; /**< Number of pages by sector in the MTD */
     uint32_t page_size;        /**< Size of the pages in the MTD */
+    uint32_t write_size;       /**< Minimum size and alignment of writes to the device
+                                 *
+                                 * If this is zero, it means that the flash
+                                 * driver did not specify a write size. Users
+                                 * should not guess, but fix their drivers;
+                                 * eventually, initializing an MTD with a write
+                                 * size of 0 will become an error. */
```

Comment:
Now set to page size (as that for sure can be written in one go, as it is the maximum that can).

Reply:
For these users I think that the new behavior (finding 0, setting pagesize) does the right thing: That user should just set `write_gran = mtd->write_size` (or `* 8`, if that's supposed to be bits).

That may easily result in runtime errors if FlashDB expects that to be smaller than some size. (Tried to find a pointer, but from a brief GitHub search ... it appears that write_gran isn't even ever read?) These errors would be justified: As long as nobody has entered the information into the system that this device supports byte granularity, there is no way to tell, and whoever builds something based on FlashDB will need to enhance the driver they use to set the correct size.

We can't just assume that every driver that doesn't set it has a write size of 1 -- mtd_flashpage didn't set it and that most certainly does not have a write size of 1. (It may easily be that that is the *only* MTD driver that has a write size > 1, but I don't know them all well enough to tell, and even if anyone were to go through all the datasheets of implemented drivers, people could have private MTD modules we don't find).

Reply:
Then let's make it

    assert)mtd->write_size);

so it blows up directly instead of producing code that fails somewhere down the line or is just unnecessarily slow and hard to debug.

Reply:
Yes, that'd work, fixup'ing.

Still breaks applications ("needlessly" if they never access write_size, but then again chances are that's just because they're not making their assumptions explicit), and as you say it'd be easy to debug.